includes/polling/os/pmp.inc.php

```diff
@@ -61,3 +61,192 @@
+if (is_numeric($visible) && is_numeric($tracked)) {
+    $rrd_def = RrdDefinition::make()
+        ->addDataset('visible', 'GAUGE', 0, 1000)
+        ->addDataset('tracked', 'GAUGE', 0, 1000);
+    $fields = array(
+        'visible' => floatval($visible),
+        'tracked' => floatval($tracked),
+    );
+    $tags = compact('rrd_def');
+    data_update($device, 'canopy-generic-gpsStats', $tags, $fields);
+    $graphs['canopy_generic_gpsStats'] = true;
+    unset($rrd_filename, $visible, $tracked);
+}
+
+$dbmRadio = str_replace('"', "", snmp_get($device, "radioDbmInt.0", "-Ovqn", "WHISP-SM-MIB"));
+$minRadio = str_replace('"', "", snmp_get($device, "minRadioDbm.0", "-Ovqn", "WHISP-SM-MIB"));
+$maxRadio = str_replace('"', "", snmp_get($device, "maxRadioDbm.0", "-Ovqn", "WHISP-SM-MIB"));
+$avgRadio = str_replace('"', "", snmp_get($device, "radioDbmAvg.0", "-Ovqn", "WHISP-SM-MIB"));
+
+if (is_numeric($dbmRadio) && is_numeric($minRadio) && is_numeric($maxRadio) && is_numeric($avgRadio)) {
+    $rrd_def = RrdDefinition::make()
+        ->addDataset('dbm', 'GAUGE', -100, 0)
+        ->addDataset('min', 'GAUGE', -100, 0)
+        ->addDataset('max', 'GAUGE', -100, 0)
+        ->addDataset('avg', 'GAUGE', -100, 0);
+
+    $fields = array(
+        'dbm' => $dbmRadio,
+        'min' => $minRadio,
+        'max' => $maxRadio,
+        'avg' => $avgRadio,
+    );
+    $tags = compact('rrd_def');
+    data_update($device, 'canopy-generic-radioDbm', $tags, $fields);
+    $graphs['canopy_generic_radioDbm'] = true;
+    unset($rrd_filename, $dbmRadio, $minRadio, $maxRadio, $avgRadio);
+}
+
+$horizontal = str_replace('"', "", snmp_get($device, "linkRadioDbmHorizontal.2", "-Ovqn", "WHISP-APS-MIB"));
+$vertical = str_replace('"', "", snmp_get($device, "linkRadioDbmVertical.2", "-Ovqn", "WHISP-APS-MIB"));
+if (is_numeric($horizontal) && is_numeric($vertical)) {
+    $rrd_def = RrdDefinition::make()
+        ->addDataset('horizontal', 'GAUGE', -100, 0)
+        ->addDataset('vertical', 'GAUGE', -100, 0);
+    $fields = array(
+        'horizontal' => $horizontal,
+        'vertical' => $vertical,
+    );
+    $tags = compact('rrd_def');
+    data_update($device, 'canopy-generic-450-linkRadioDbm', $tags, $fields);
+    $graphs['canopy_generic_450_linkRadioDbm'] = true;
+    unset($rrd_filename, $horizontal, $horizontal);
+}
+
+$lastLevel = str_replace('"', "", snmp_get($device, "lastPowerLevel.2", "-Ovqn", "WHISP-APS-MIB"));
+if (is_numeric($lastLevel)) {
+    $rrd_def = RrdDefinition::make()->addDataset('last', 'GAUGE', -100, 0);
+    $fields = array(
+        'last' => $lastLevel,
+    );
+    $tags = compact('rrd_def');
+    data_update($device, 'canopy-generic-450-powerlevel', $tags, $fields);
+    $graphs['canopy_generic_450_powerlevel'] = true;
+    unset($lastLevel);
+}
+
+$ssr = str_replace('"', "", snmp_get($device, "linkSignalStrengthRatio.2", "-Ovqn", "WHISP-APS-MIB"));
```

Comment:
Is SSR a form of SNR?

Reply:
Here's the desc as per the mib: 
 Signal Strength Ratio in dB is the power received by the vertical antenna input (dB) -
power received by the horizontal antenna input (dB).
MIMO radios only.

I'm not sure, it might be a bit of a stretch but could potentially work putting it in snr.

Reply:
It's not an SNR, it's a measure of the imbalance in the chains. It's actually calculable and useful for all dual polarity radios, and presented directly in many devices. A change in the SSR is indicative of movement of an antenna in one plane only, or of degredation in cables or antenna affecting one polarity only. Because of the opaque nature of the calculations done on multi chain radios to generate an SNR figure, a degredation in one chain only can be masked or completely hidden in the headline figure.

This highlights one of the issues I see with the new wireless sensors paradigm, enterprise grade radios have a lot of metrics which are either not relevant to or not easily measureable on, consumer WiFi products. With regard to cambium, the new Medusa radios have further metrics relating to the active phased array antenna. Similar metrics will also be present on the Radwin Jet and probably future ubiquiti and Mikrotik radios.

Reply:
@barryodonovan I have no problem with adding metrics.

Reply:
I'll throw in a new metric!